pyth_observer/calendar.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+import pytz
+
+TZ = pytz.timezone('America/New_York')
+
+EQUITY_OPEN = datetime.time(9, 30, 0, tzinfo=TZ)
+EQUITY_CLOSE = datetime.time(16, 0, 0, tzinfo=TZ)
+EQUITY_EARLY_CLOSE = datetime.time(13, 0, 0, tzinfo=TZ)
+EQUITY_HOLIDAYS = [
+    datetime.datetime(2022, 1, 17, tzinfo=TZ).date(),
+    datetime.datetime(2022, 2, 8, tzinfo=TZ).date(),
+    datetime.datetime(2022, 2, 21, tzinfo=TZ).date(),
+    datetime.datetime(2022, 4, 15, tzinfo=TZ).date(),
+    datetime.datetime(2022, 5, 30, tzinfo=TZ).date(),
+    datetime.datetime(2022, 6, 20, tzinfo=TZ).date(),
+    datetime.datetime(2022, 7, 4, tzinfo=TZ).date(),
+    datetime.datetime(2022, 9, 5, tzinfo=TZ).date(),
+    datetime.datetime(2022, 11, 24, tzinfo=TZ).date(),
+    datetime.datetime(2022, 11, 25, tzinfo=TZ).date(),
+    datetime.datetime(2022, 12, 26, tzinfo=TZ).date()
+]
+EQUITY_EARLY_HOLIDAYS = [
+    datetime.datetime(2022, 2, 8, tzinfo=TZ).date(),
+    datetime.datetime(2022, 11, 25, tzinfo=TZ).date()
+]
+
+
+class Calendar():
+    def is_market_open(self, product, dt):
+        # equity market
+        if product.attrs['asset_type'] == 'Equity':
+            day, date, time = dt.weekday(), dt.date(), dt.time()
+            if date in EQUITY_HOLIDAYS:
+                if date in EQUITY_EARLY_HOLIDAYS and time >= EQUITY_OPEN and time <= EQUITY_EARLY_CLOSE:
+                    print("Hello")
```

Comment:
```suggestion
```

Reply:
^ definitely do this ,

also I think it would be preferable if the dates in EQUITY_EARLY_HOLIDAYS didn't need to appear in EQUITY_HOLIDAYS. The current setup creates the potential for a configuration bug where someone adds to EQUITY_EARLY_HOLIDAYS but forgets EQUITY_HOLIDAYS.